### ISSUE TITLE
datadog exporter ignores attributes

### ISSUE THREAD

**Issue by deweyjose** [apollographql/router#2066]
**Describe the bug**
The Datadog exporter does not apply attributes configured in Router.yaml to spans. Our team needs to set the `version` tag at the root of all traces exported to Datadog for Deployment tracking.

**To Reproduce**

```
  tracing:
    trace_config:
      service_name: "${env.DD_SERVICE:-rusty-router}"
      attributes:
        version: "${env.DD_VERSION:-test}"
    propagation:
      datadog: true
```

**Expected behavior**
We expect to all traces to include a version tag at the root of each span named version.

**Output**
If applicable, add output to help explain your problem.
We do not see the version tag.

**Desktop (please complete the following information):**

**Additional context**

As a workaround we added a supergraph plugin that adds a span explicitly for this purpose, but it seems wasteful. It also does not help when Router receives invalid gql request that cause 400s as those short circuit any plugins we right - our version spans will not be applied.

**Comment by bnjjj** [comment]
related to https://github.com/apollographql/router/issues/1162

**Comment by BrynCooke** [comment]
This is fixed in the otel upgrade branch but we need a new otel release.

**Comment by abernix** [comment]
I'll put the same comment here as I put on https://github.com/apollographql/router/issues/1162#issuecomment-1473557288:

> This is now waiting on open-telemetry/opentelemetry-rust#965, which we're hoping comes soon.

(We are led to believe it's coming very soon. There's just maybe a patch in there somewhere that might be very relevant for Datadog, specifically.)

**Comment by abernix** [comment]
The fix for this is purportedly is now in https://github.com/open-telemetry/opentelemetry-rust/releases/tag/v0.19.0.   I'll see about getting the integration of that upgrade prioritized on our side.

**Comment by jmvtrinidad** [comment]
> The fix for the DD exporter is in an unreleased version of open-telemetry-datadog, but if you're building a custom binary you can apply a patch in your Cargo.toml:
> 
> https://github.com/apollographql/router/blob/3bb8a67e16c2d8beb517ec2139385820e6d63e8d/Cargo.toml#L43-L49
> 
> I confirmed that the patches work.
> 
> An alternative is to configure your DD agent to listen for OTLP traces.
> 
> ```yaml
> # datadog agent config
> otlp_config:
>   receiver:
>     protocols:
>       grpc:
>       endpoint: 0.0.0.0:4317
> ```
> 
> ```yaml
> # router config
> telemetry:
>   tracing:
>     trace_config:
>       attributes:
>         version: "2"
>         env: "dev"
>         test: "42"
>     otlp:
>       endpoint: 0.0.0.0:4317
>       protocol: grpc
> ```

@lennyburdette do you mean to uncomment the patch.crates-io and build? Thanks

**Comment by abernix** [comment]
We are unfortunately still blocked on the OpenTelemetry update because of upstream dependencies.  It is [getting close](https://github.com/tokio-rs/tracing-opentelemetry/pull/12) though.

**Comment by abernix** [comment]
Ok, in theory, this is fixed by https://github.com/tokio-rs/tracing-opentelemetry/releases/tag/v0.19.0 and we have an umbrella ticket https://github.com/apollographql/router/issues/2878 that tracks that need (since it unblocks many constraints — like this one — now, at least in theory.)  We're proritizing making that update in the next week or two.

**Comment by garypen** [comment]
I have verified that this is fixed in my opentelemetry 0.19.0 PR: #3196
